docs/book/profiler.md

```diff
@@ -0,0 +1,257 @@
+# Profiler
+
+This page describes how to use the code profiler for Rego. The Profiler
+captures the following information about Rego expressions:
+
+  * Expression Time
+  * Number of Evaluations
+  * Number of Re-Evaluations (Redo)
+  * Expression Location in source code
+
+OPA's `eval` command provides the following profiler options
+
+| Command line Option        | Detail                                                        | Default |
+|:--------------------------:|:-------------------------------------------------------------:|:-----:|
+| `profile`                  | Returns profiling results sorted by the default or given criteria              | NA |
+| `profile-sort`             | Criteria to sort the expression profiling results. This options implies `--profile`. |  total_time_ns => num_eval => num_redo => file => line |
+| `profile-limit`            | Desired number of profiling results sorted on the given criteria. This options implies `--profile`. |  10 |
+
+## Sort criteria for the profile results
+  * `total_time_ns` - Results are displayed is decreasing order of _expression evaluation time_
+  * `num_eval`  - Results are displayed is decreasing order of _number of times an expression is evaluated_
+  * `num_redo`  - Results are displayed is decreasing order of _number of times an expression is re-evaluated(redo)_
+  * `file`  - Results are sorted in reverse alphabetical order based on the _rego source filename_
+  * `line`  - Results are displayed is decreasing order of _expression line number_ in the source file
+
+When the sort criteria is not provided `total_time_ns` has the **highest** priority
+while `line` has the **lowest**.
+
+#### Example Policy
+The different profiling examples shown later on this page use the below
+sample policy.
+
+{%ace edit=true, lang='python'%}
```

Comment:
Can we put this inside a fenced code block or will it break the plugin? It's nice to be able to have the docs rendered relatively well in GitHub...

Reply:
code blocks break the plugin.